crates/conformance-test-utils/src/madara.rs

```diff
@@ -0,0 +1,154 @@
+    // Compute the storage key for the storage variable name and the keys.
+    let mut storage_key = get_storage_var_address(storage_variable_name, keys)?;
+
+    // Add the offset to the storage key.
+    storage_key += FieldElement::from(storage_key_offset);
+
+    let contract_address: ContractAddress = starknet_address.into();
+
+    // Create the tuple for the initial storage data on the Kakarot contract with the given storage key.
+    let storage_data = ((contract_address, storage_key.into()), storage_value.into());
+
+    Ok(storage_data)
+}
+
+/// Generates the genesis storage tuples for pre-funding a Starknet address on Starknet.
+///
+/// This function calculates the storage keys for the balance of the ERC20 Fee Token
+/// contract using the provided Starknet address. The resulting Vec of tuples represent the initial
+/// storage of the Fee Token contract, where the account associated with the Starknet address is
+/// pre-funded with the specified `amount`. The `amount` is split into two 128-bit chunks, which
+/// are stored in the storage keys at offsets 0 and 1.
+pub fn genesis_fund_starknet_address(
+    starknet_address: FieldElement,
+    amount: U256,
+) -> Result<Vec<((ContractAddress, StorageKey), StorageValue)>> {
+    // Split the amount into two 128-bit chunks.
+    let low = amount & U256::from(U128::MAX);
+    let high = amount >> 128;
+
+    // The storage key offsets for the two 128-bit chunks.
+    let uint256_offset = [(low, 0), (high, 1)]; // (value, offset)
+
+    // Iterate over the storage key offsets and generate the storage tuples.
+    uint256_offset
+        .iter()
+        .filter(|(value, _)| *value != U256::from(0)) // Only add non-zero balances.
+        .map(|(value, offset)| {
+            genesis_set_storage_starknet_contract(
+                FieldElement::from_hex_be(STARKNET_NATIVE_TOKEN)?,
+                "ERC20_balances",
+                &[starknet_address],
+                FieldElement::from_bytes_be(&value.to_be_bytes())?,
+                *offset,
+            )
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
+
+    use kakarot_rpc_core::client::constants::STARKNET_NATIVE_TOKEN;
+    use kakarot_rpc_core::mock::constants::ACCOUNT_ADDRESS;
+    use reth_primitives::{U128, U256};
+    use starknet::core::types::FieldElement;
+    use starknet::core::utils::get_storage_var_address;
+
+    use super::{genesis_fund_starknet_address, genesis_set_storage_starknet_contract};
+
+    /// This test verifies that the `genesis_set_storage_starknet_contract` function generates the
+    /// correct tuples for a given Starknet address, storage variable name, keys, storage value,
+    /// and storage key offset.
+    #[tokio::test]
+    async fn test_genesis_set_storage_starknet_contract() {
+        // Given
+        let starknet_address = *ACCOUNT_ADDRESS;
+        let storage_variable_name = "_signer";
+        let keys = vec![];
```

Comment:
this is a random choice right?

Reply:
Yeah just a random one, I wanted to show that it can be something apart from `ER20_balances` and `storage_`

Reply:
so also replace the `ACCOUNT_ADDRESS` plz